public/src/components/channelManagement/CampaignSelector.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react';
+import { fetchFrontendSettings, FrontendSettingsType } from '../../utils/requests';
+import { DataFromServer } from '../../hocs/withS3Data';
+import { Campaign } from './campaigns/CampaignsForm';
+import { Test } from './helpers/shared';
+
+import { Select, MenuItem, FormControl, makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(() => ({
+  dialogHeader: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    paddingRight: '8px',
+  },
+  input: {
+    '& input': {
+      textTransform: 'uppercase !important',
+    },
+  },
+  campaignSelector: {
+    marginBottom: '20px',
+    marginRight: '12px',
+    minWidth: '200px',
+  },
+  campaignSelectorContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  warning: {
+    color: '#555',
+  },
+}));
+
+interface CampaignSelectorProps {
+  onCampaignChange: (campaign?: string) => void;
+  test: Test;
+  disabled: boolean;
+}
+
+const CampaignSelector: React.FC<CampaignSelectorProps> = ({
+  onCampaignChange,
+  test,
+  disabled,
+}: CampaignSelectorProps) => {
+  const classes = useStyles();
+
+  const [campaigns, setCampaigns] = useState<Campaign[]>([]);
+
+  useEffect(() => {
+    fetchFrontendSettings(FrontendSettingsType.campaigns).then((data: DataFromServer<Campaign[]>) =>
+      setCampaigns(data.value),
+    );
+  }, []);
+
+  const setCampaignName = (campaign?: string) => onCampaignChange(campaign);
+
+  return (
+    <>
+      <FormControl className={classes.campaignSelector}>
+        <Select
+          value={test.campaignName}
+          displayEmpty
+          renderValue={(campaign): string => {
+            if (campaign === undefined) {
+              return ''; // triggers the displayEmpty behaviour
+            }
+            return campaign as string;
+          }}
+          onChange={(event: React.ChangeEvent<{ value: unknown }>): void => {
+            setCampaignName(event.target.value as string | undefined);
+          }}
+          disabled={disabled}
+        >
+          <MenuItem value={undefined} key={'campaignName-none'}>
+            No campaign
+          </MenuItem>
+          {campaigns.map(campaign => (
+            <MenuItem value={campaign.name} key={`campaign-${campaign.name}`}>
+              {campaign.name}
+            </MenuItem>
+          ))}
+        </Select>
+      </FormControl>
+      <div className={classes.warning}>
+        If you want to use this Test in a new or different campaign, please consider copying it,
+        rather than changing the Test campaign value.
```

Comment:
what's the reason for this message? It takes up a lot of room!

Reply:
I was concerned about moving tests in a different campaign when the name of the original campaign is as a prefix in the test name. This message is to encourage good practices.
But it's true that the whole campaign part takes quite a lot of space.

Reply:
Message added in line with Mael's comments. Message moved to sit to the right of the selector box.